### PR TITLE
Added connection timeout to ssh command in pi_local.sh.

### PR DESCRIPTION
**Pull request by jasondoze**
Renamed install_vault to vault_install.

Add loops to prompt for secrets, 5 attempts for 3 keys.

Add vault_secrets to put and get secrets from json file.

Add test.json for example secrets to put and get.

Next, the README.md

### CODE REVIEW

File: test.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  }
+],
+"config": {
+  "max_connections": 100,
+  "timeout": 60,
+  "logging_enabled": true
+},
+"secrets": {
+  "database": {
+    "username": "bingo_gringo",
```
Comment (stephenlauck) [inline test.json:21]: lol

File: vault_install.sh [hunk 2]
```diff
@@ -1,7 +1,15 @@
 #!/bin/bash
 
 # This script updates the package list, installs Gpg, downloads the Hashicorp signing key, adds the Hashicorp repository and installs Vault.
-sudo apt update
+
+# Update Apt
+if (( $(date +%s) - $(stat -c %Y /var/lib/apt/periodic/update-success-stamp) > 24*60*60 ))
```
Comment (stephenlauck) [inline vault_install.sh]: What are you doing here?

File: vault_install.sh [hunk 3]
```diff
@@ -28,7 +36,8 @@ then
   echo -e "\n\033[1;32m==== Hashicorp repo present ====\033[0m\n"
 else
   echo -e "\n\033[1;33m==== Adding Hashicorp repo ====\033[0m\n"
-  echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] https://apt.releases.hashicorp.com $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/hashicorp.list
+  echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] \
+  https://apt.releases.hashicorp.com $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/hashicorp.list
   sudo apt update
```
Comment (stephenlauck) [inline vault_install.sh:49]: If you have the apt update above, why are you doing this here? Do you know about the apt-cache? Could you just check if the package is in the apt-cache and update if not?

File: vault_install.sh (hunk 2 above)
Comment (jasondoze) [reply vault_install.sh]: I looked at several ways to check if apt had been updated. This version seemed to be popular on stack overflow. Folks were commenting that the cache and apt/lists weren’t reliable methods to determine if it had been updated but idk.

File: vault_install.sh (hunk 3 above)
Comment (jasondoze) [reply vault_install.sh:49]: This is from the Linux vault install docs. Maybe I’m misunderstanding how this works.

**Review by stephenlauck — APPROVED** [review]

**Pull request merged**